### ISSUE TITLE
Work around for incorrect metadata encoding of structs with delegate fields

### ISSUE THREAD

**Issue by rylev** [microsoft/windows-rs#444]
Trying to build the following bindings using either v0.2.1 or the master branch of the windows crate leads to an error:
```rust
fn main() {
    windows::build!(
        windows::win32::projected_file_system::*
    );
}
```

```
error[E0599]: no method named `map` found for raw pointer `*mut Option<extern "system" fn(*const PRJ_CALLBACK_DATA, u64, u32) -> ErrorCode>` in the current scope
   --> C:\Users\ryanl\test\target\debug\build\test-e6d01d4d62acab2d\out/windows.rs:474:56
    |
474 |                         && self.get_file_data_callback.map(|f| f as usize)
    |                                                        ^^^ method not found in `*mut Option<extern "system" fn(*const PRJ_CALLBACK_DATA, u64, u32) -> ErrorCode>`
    |
    = note: try using `<*const T>::as_ref()` to get a reference to the type behind the pointer: https://doc.rust-lang.org/std/primitive.pointer.html#method.as_ref
    = note: using `<*const T>::as_ref()` on a pointer which is unaligned or points to invalid or uninitialized memory is undefined behavior
    = note: the method `map` exists but the following trait bounds were not satisfied:
            `*mut Option<extern "system" fn(*const PRJ_CALLBACK_DATA, u64, u32) -> ErrorCode>: Iterator`
            which is required by `&mut *mut Option<extern "system" fn(*const PRJ_CALLBACK_DATA, u64, u32) -> ErrorCode>: Iterator`
```

**Comment by rylev** [comment]
The code in question is being generated [here](https://github.com/microsoft/windows-rs/blob/256538e85f2f9c860c09c318423b3be79b36b9c3/crates/gen/src/struct.rs#L209) inside the definition for `PartialEq` for structs. The `TypeKind` is `TypeKind::Delegate` and definition is not `winrt`. 

It seems like the fix is just dereferencing the pointers.

**Comment by rylev** [comment]
Whoops this is a duplicate of #438. @kennykerr can we keep this open as a tracking issue even if it's blocked on work outside our control?

**Comment by kennykerr** [comment]
For reference: https://github.com/microsoft/win32metadata/issues/132

**Comment by kennykerr** [comment]
There is now a workaround in `windows-rs` for this issue.

**Comment by rylev** [comment]
The work around for this can be found [here](https://github.com/microsoft/windows-rs/blob/2a8e603495dc9e1f68955a54226c2502acf8f147/crates/gen/src/struct.rs#L30-L33). I'm going to rename this issue to better reflect its current state.

**Comment by kennykerr** [comment]
This was fixed recently.
